File: app/src/main/java/io/weimu/www/imagepicker/activity/PhotoViewPagerActivity.java
package io.weimu.www.imagepicker.activity;

import android.content.Context;
import android.content.Intent;
import android.support.v4.view.ViewPager;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.WindowManager;

import java.util.ArrayList;
import java.util.List;

import io.weimu.www.imagepicker.R;
import io.weimu.www.imagepicker.activity.base.BaseActivity;
import io.weimu.www.imagepicker.adaper.pageradapter.PreviewViewPager;
import io.weimu.www.imagepicker.adaper.pageradapter.base.BaseFragmentPagerAdapter;
import io.weimu.www.imagepicker.fragment.ImagePreviewFragment;
import io.weimu.www.imagepicker.fragment.base.BaseFragment;

public class PhotoViewPagerActivity extends BaseActivity {
    private PreviewViewPager mViewPager;
    private Toolbar toolbar;
    @Override
    protected void findViewByIDS() {
        mViewPager = myFindViewsById(R.id.id_vp);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_photo_view_pager;
    }


    private boolean isShowBar = true;

    private List<String> imagList = new ArrayList<String>();
    private List<BaseFragment> fragments = new ArrayList<>();
    private int position = 0;

    public static final String POSITION = "position";
    public static final String PICLIST = "piclist";
    private BaseFragmentPagerAdapter mAdapter;

    public static Intent newInstance(Context context, int position, ArrayList<String> imagList) {
        Intent i = new Intent(context, PhotoViewPagerActivity.class);
        i.putExtra(POSITION, position);
        i.putStringArrayListExtra(PICLIST, imagList);
        return i;
    }

    @Override
    protected void onGenerate() {
        initFirst();
        initToolBar();
        initViewPager();
    }

    private void initToolBar() {
        toolbar = (Toolbar) findViewById(com.yongchun.library.R.id.toolbar);
        toolbar.setTitle((position + 1) + "/" + imagList.size());
        setSupportActionBar(toolbar);
        toolbar.setNavigationIcon(com.yongchun.library.R.mipmap.ic_back);
    }

    private void initFirst() {
        position = getIntent().getIntExtra(POSITION, 0);
        imagList = getIntent().getStringArrayListExtra(PICLIST);
    }

    private void initViewPager() {
        for (int i = 0; i < imagList.size(); i++) {
            fragments.add(ImagePreviewFragment.newInstance(imagList.get(i)));
        }
        mAdapter = new BaseFragmentPagerAdapter(getSupportFragmentManager(), fragments);
        mViewPager.setAdapter(mAdapter);
        mViewPager.setCurrentItem(position);
        mViewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                toolbar.setTitle((position + 1) + "/" + imagList.size());
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });


    }


    private void hideStatusBar() {
        WindowManager.LayoutParams attrs = getWindow().getAttributes();
        attrs.flags |= WindowManager.LayoutParams.FLAG_FULLSCREEN;
        getWindow().setAttributes(attrs);
    }

    private void showStatusBar() {
        WindowManager.LayoutParams attrs = getWindow().getAttributes();
        attrs.flags &= ~WindowManager.LayoutParams.FLAG_FULLSCREEN;
        getWindow().setAttributes(attrs);
    }

    public void switchBarVisibility() {
        toolbar.setVisibility(isShowBar ? View.GONE : View.VISIBLE);
        if (isShowBar) {
            hideStatusBar();
        } else {
            showStatusBar();
        }
        isShowBar = !isShowBar;
    }
}

File: app/src/main/java/io/weimu/www/imagepicker/activity/PhotoViewPagerActivity.kt
package io.weimu.www.imagepicker.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.FragmentManager
import android.support.v4.view.ViewPager
import android.view.View
import android.view.WindowManager
import com.shizhefei.view.largeimage.LargeImageView
import com.weimu.library.widget.DragViewPager
import com.weimu.universalib.ktx.getColorPro
import com.weimu.universalib.ktx.getStatusBarHeight
import com.weimu.universalview.core.activity.BaseActivity
import com.weimu.universalview.core.fragment.BaseFragment
import com.weimu.universalview.core.pager.BaseFragmentStatePagerAdapter
import com.weimu.universalview.core.toolbar.StatusBarManager
import com.weimu.universalview.core.toolbar.ToolBarManager
import com.weimu.universalview.ktx.setMargins
import io.weimu.www.imagepicker.R
import java.util.*
import kotlin.properties.Delegates

class PhotoViewPagerActivity : BaseActivity() {


    companion object {

        val POSITION = "position"
        val PICLIST = "piclist"

        fun newInstance(context: Context, position: Int, imagList: ArrayList<String>): Intent {
            val i = Intent(context, PhotoViewPagerActivity::class.java)
            i.putExtra(POSITION, position)
            i.putStringArrayListExtra(PICLIST, imagList)
            return i
        }
    }

    private val mViewPager: DragViewPager by lazy { findViewById<DragViewPager>(R.id.id_vp) }

    var isShowBar by Delegates.observable(false) { property, oldValue, newValue ->
        //todo是否显示
        if (newValue) {
            showStatusBar()
        } else {
            hideStatusBar()
        }
        if (newValue) {
            toolBarManager.showToolBar()
        } else {
            toolBarManager.hideToolBar()
        }
    }

    private var imagList: List<String> = ArrayList()
    private val fragments = ArrayList<BaseFragment>()
    private var position = 0
    private lateinit var mAdapter: ImagePagerAdapter

    private lateinit var toolBarManager: ToolBarManager


    override fun getLayoutResID(): Int = R.layout.activity_photo_view_pager

    override fun afterViewAttach(savedInstanceState: Bundle?) {
        initData()
        initToolBar()
        initViewPager()
    }


    private fun initData() {
        position = intent.getIntExtra(POSITION, 0)
        imagList = intent.getStringArrayListExtra(PICLIST)
    }


    private fun initToolBar() {
        StatusBarManager.setColor(window, getColorPro(R.color.white))
        StatusBarManager.setLightMode(window)
        toolBarManager = ToolBarManager.with(this, getContentView())
                .bg {
                    this.setBackgroundResource(com.weimu.library.R.color.white)
                }
                .leftMenuIcon {
                    this.setImageResource(com.weimu.library.R.drawable.toolbar_arrow_back_black)
                }
                .title {
                    this.text = "${(position + 1).toString() + "/" + imagList.size}"
                }
    }


    private fun initViewPager() {
        mViewPager.apply {
            for (i in imagList.indices) {
                fragments.add(ImagePreviewAppFragment.newInstance(imagList[i]))
            }
            mAdapter = ImagePagerAdapter(supportFragmentManager)
            this.adapter = mAdapter
            mAdapter.setFragments(fragments)
            this.currentItem = position
            this.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
                override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

                }

                override fun onPageSelected(position: Int) {
                    toolBarManager.title {
                        this.text = (position + 1).toString() + "/" + imagList.size
                    }
                }

                override fun onPageScrollStateChanged(state: Int) {

                }
            })
            //触发接口
            this.setIAnimClose(object : DragViewPager.IAnimClose {
                override fun onPictureClick() {

                }

                override fun onPictureRelease(view: View?) {
                    onBackPressed()
                }

            })


        }

    }


    private fun hideStatusBar() {
        val attrs = window.attributes
        attrs.flags = attrs.flags or WindowManager.LayoutParams.FLAG_FULLSCREEN
        window.attributes = attrs
    }

    private fun showStatusBar() {
        val attrs = window.attributes
        attrs.flags = attrs.flags and WindowManager.LayoutParams.FLAG_FULLSCREEN.inv()
        window.attributes = attrs
    }


    class ImagePagerAdapter(fm: FragmentManager) : BaseFragmentStatePagerAdapter(fm), DragViewPager.DragViewAdapter {
        override fun getImageView(position: Int): LargeImageView = getItem(position).view!!.findViewById(R.id.iv_large)
    }


}
